onnxruntime/test/fuzzing/src/test.cpp

```diff
@@ -285,123 +280,113 @@ int main(int argc, char* argv[])
+          std::chrono::seconds timeInSec{testTimeOut};
+          std::chrono::hours timeInHrs{testTimeOut};
+          std::chrono::system_clock::time_point endTime{currTime};
+          endTime += scale == timeScale::Hrs ? timeInHrs
+                                             : scale == timeScale::Min ? timeInMin : timeInSec;
+          Logger::testLog << "Starting Test" << Logger::endl;
+          while (currTime < endTime) {
+            try {
+              onnx::ModelProto bad_model = model_proto;
+              Logger::testLog << "Starting Test iteration: " << iteration << Logger::endl;
+              mutateModelTest(bad_model, mutateModelDirName, userOpt);
+              num_successful_runs++;
+              Logger::testLog << "Completed Test iteration: " << iteration++ << Logger::endl;
+            } catch (...) {
+              fuzz_handle_exception();
+            }
+            // Update current time
+            //
+            currTime = std::chrono::system_clock::now();
+          }
         }
-
-        Logger::testLog<< L"Total number of exceptions: " << num_unknown_exception+num_std_exception+num_ort_exception << Logger::endl;
-        Logger::testLog<< L"Number of Unknown exceptions: " << num_unknown_exception << Logger::endl;
-        Logger::testLog<< L"Number of ort exceptions: " << num_ort_exception << Logger::endl;
-        Logger::testLog<< L"Number of std exceptions: " << num_std_exception << Logger::endl;
-        Logger::testLog << L"Number of unique errors: "<< werrStreamBuf.get_unique_errors() << L"\n";
-        
-        if(userOpt.stress)
-        {
-          Logger::testLog.disable();
-          Logger::testLog.flush();
+      } else {
+        throw std::exception("Unable to initialize the Onnx model in memory");
+      }
+    } else {
+      std::wstring ortModelFile = modelFile + L".ort";
+      Ort::SessionOptions so;
+      Ort::Env env;
+      so.SetGraphOptimizationLevel(ORT_DISABLE_ALL);
+      so.SetOptimizedModelFilePath(ortModelFile.c_str());
+      so.AddConfigEntry(kOrtSessionOptionsConfigSaveModelFormat, "ORT");
+      Ort::Session session(env, modelFile.c_str(), so);
+      size_t numBytes = std::filesystem::file_size(ortModelFile);
+      std::vector<char> modelData(numBytes);
+      std::ifstream ortModelStream(ortModelFile, std::ifstream::in | std::ifstream::binary);
+      ortModelStream.read(modelData.data(), numBytes);
+      ortModelStream.close();
+      for (int i = 8; i < numBytes; i++) {
+        char tmp = modelData[i];
+        int j = (numBytes - i - 1) ? (i + 1) : 0;  // j = (i+1) % (numBytes -1)
+        modelData[i] ^= modelData[j];
+        modelData[i] = tmp;
+        try {
+          Logger::testLog << "Starting Test" << Logger::endl;
+          OnnxPrediction predict(modelData);
+          predict.SetupInput(GenerateDataForInputTypeTensor, 0);
+          predict.RunInference();
+          predict.PrintOutputValues();
+          num_successful_runs++;
```

Comment:
What's the reason for printing output values?

Reply:
My understanding is, the test runs through all the mutations capturing all the exceptions whether or not handled by onnxruntime and finally produces a the summary, which may inspected to know whether or not all mutations generated any errors.

---
In reply to: [521692691](https://github.com/microsoft/onnxruntime/pull/5771#discussion_r521692691) [](ancestors = 521692691)

Reply:
This was in the original code. Yes it dumps large tensors to the stdout. Unless it is used, I will go ahead and remove it.